### PR TITLE
model summary: detect NaN objective function values from bbi

### PR DESCRIPTION
**Pull request by kyleam**
If the lst files contain values like

    OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT:                        NaN
    OBJECTIVE FUNCTION VALUE WITH CONSTANT:                           NaN

then summarizing the model fails with

    > model_summary(mod)
    Error: parse error: premature EOF

                         (right here) ------^

The main fix for this is on the bbi side [*].  With that change, NaN
values now come through as DefaultFloat64 (aka BBI_NULL_NUM).  Teach
create_summary_object() to map BBI_NULL_NUM to NA.

Making bbr flag these values is left for later (gh-505).

[*] https://github.com/metrumresearchgroup/bbi/pull/268

---

Tests pass for me locally with a bbi built from 5561ec0 (nmparser: use strToFloat to handle NaN objective function values, 2022-06-23):

```
── Skipped tests  ──────────────────────────────────────────────────────────────
• getOption("bbr.DEV_no_min_version") is TRUE (1)

[ FAIL 0 | WARN 0 | SKIP 1 | PASS 984 ]
```

There are other incoming bbi changes, so I think it'd make sense to hold off tagging a bbi dev release and testing on bbr's Drone until more of those changes have landed.

### CODE REVIEW

File: tests/testthat/test-model-summary.R [hunk 1]
```diff
@@ -183,4 +183,27 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
...
+    skip_if_old_bbi("3.1.1")
+    withr::with_tempdir({
+      fs::dir_copy(MOD1_ABS_PATH, "tmpmod")
+      mod <- copy_model_from(read_model(MOD1_ABS_PATH),
+                             file.path(getwd(), "tmpmod"))
+
+      lst_file <- file.path("tmpmod", "1.lst")
+      lst_lines <- readr::read_lines(lst_file)
+      stringr::str_replace(lst_lines,
+                           "^( OBJECTIVE FUNCTION VALUE .*: +)([0-9.]+)\\s*$",
```
Comment (kyleam) [inline tests/testthat/test-model-summary.R:197]: There's no need to match a second group here, will remove.

File: tests/testthat/test-model-summary.R [hunk 2]
```diff
@@ -183,4 +183,27 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
...
+                           "\\1NaN")  %>%
+        stringr::str_replace("^( #OBJV:\\*+ +)([0-9.]+)( +\\*+\\s*)$",
+                             "\\1NaN\\1")  %>%
```
Comment (kyleam) [inline tests/testthat/test-model-summary.R:200]: This happens to work, but the second group in the replacement should be `\\3`, not `\\1`.  And, like above, there's no reason to capture the numeric value.

**Comment by kyleam** [comment]
With merge of gh-511, this PR had a conflict in `test-model-summary.R` and a semantic conflict in the test ID.  Rebased to resolve.

<details>
<summary>range-diff</summary>

```
$ git range-diff main origin/objfn-nan objfn-nan
```

```
1:  72b22df ! 1:  2d73396 model summary: detect NaN objective function values from bbi
    @@ inst/validation/bbr-requirements.yaml: SUM-R009:
     +SUM-R010:
     +  description: model_summary() handles NA objective function values
     +  tests:
    -+  - BBR-SUM-010
    ++  - BBR-SUM-011
      TDF-R001:
        description: tags_diff.bbi_model default happy path works
        tests:
    @@ inst/validation/bbr-stories.yaml: SUM-S001:

      ## tests/testthat/test-model-summary.R ##
     @@
    -
    -     expect_error(model_summary(mod2), regexp = "More than one `\\.lst` file")
    +     expect_equal(length(mod_sum$run_details$covariance_time), 2)
        })
    -+
    -+  test_that("model_summary() maps objective function fallback to NA [BBR-SUM-010]", {
    +
    ++  test_that("model_summary() maps objective function fallback to NA [BBR-SUM-011]", {
     +    skip_if_old_bbi("3.1.1")
     +    withr::with_tempdir({
     +      fs::dir_copy(MOD1_ABS_PATH, "tmpmod")
    @@ tests/testthat/test-model-summary.R
     +    })
     +  })
      }) # closing withr::with_options
    +
    +
```

</details>

**Comment by seth127** [comment]
> we can tag alpha.2 when needed and update .drone.yml in this PR to use that, but I'll hold off until this PR is actively under review.

@kyleam I'm hoping to look at this (and https://github.com/metrumresearchgroup/bbi/pull/268 ) in the next couple days. Are we ready to do that bbi dev tag and make the drone change, or are we waiting for something else?

**Comment by kyleam** [comment]
> @kyleam I'm hoping to look at this (and https://github.com/metrumresearchgroup/bbi/pull/268 ) in the next couple days. 

Great, thanks.

> Are we ready to do that bbi dev tag and make the drone change, or are we waiting for something else?

Nope, at the time just wanted to see if anything else would land, but realistically this will be the next thing to go in on bbi's end, so will update by this afternoon.

**Review by seth127 — APPROVED** [review]
All looks good to me. Looking through this got me thinking and I opened [this related issue](https://github.com/metrumresearchgroup/bbr/issues/520) about replacing all the `-999999999` with `NA`. 

Generally speaking, I believe in being consistent, but I think implementing that in here is the right move, whether or not we end up doing it repo-wide. Especially because valid OFV values usually _are_ large negative numbers, so having `-999999999` mean `NA` is particularly confusing for OFV (since that could plausibly be a valid value).

**Comment by kyleam** [comment]
> I opened https://github.com/metrumresearchgroup/bbr/issues/520 about replacing all the -999999999 with NA.

Thanks.  Definitely worth looking into.  I only spotted one place (before this PR) that did this mapping:
https://github.com/metrumresearchgroup/bbr/blob/f31d080e8e5fba97956fe0b1d4b8a69f1f8ec22d/R/summary-log.R#L219-L225

**Pull request merged**